### PR TITLE
Cleanup Glacier2 session helper classes

### PR DESCRIPTION
**Pull request by pepone**
This PR cleanups the Glacier2 session helper classes

* Replace some Get/Set methods by corresponding properties
* Remove the use of lock this
* Add some missing parameters to the factory constructor

### CODE REVIEW

File: csharp/src/Ice/Communicator.cs [hunk 1]
```diff
@@ -983,9 +983,9 @@ public void Destroy()
             }
 
             {
-                if (Logger != null && Logger is FileLogger)
+                if (Logger != null && Logger is FileLogger fileLogger)
```
Comment (InsertCreativityHere) [inline csharp/src/Ice/Communicator.cs:986]: The first check for `Logger != null` is redundant here. `is` will evaluate to false for null expressions anyways,

File: csharp/src/Glacier2/SessionHelper.cs [hunk 2]
```diff
@@ -12,130 +12,69 @@
...
             }
         }
 
-        /// <summary>
-        /// Returns the session's communicator object.
-        /// </summary>
-        /// <returns>The communicator.</returns>
-        public Communicator?
-        Communicator()
+        /// <summary>Gets whether there is a need an active session</summary>
```
Comment (InsertCreativityHere) [inline csharp/src/Glacier2/SessionHelper.cs:30]: Feel like there's a word or two missing in this comment.

File: csharp/src/Ice/Communicator.cs (hunk 1 above)
Comment (pepone) [reply csharp/src/Ice/Communicator.cs:986]: fixed

File: csharp/src/Glacier2/SessionHelper.cs (hunk 2 above)
Comment (pepone) [reply csharp/src/Glacier2/SessionHelper.cs]: fixed

File: csharp/src/Glacier2/SessionFactoryHelper.cs [hunk 3]
```diff
@@ -19,180 +19,234 @@ namespace ZeroC.Glacier2
...
-            _logger = logger;
-            _observer = observer;
-
-            SetDefaultProperties();
-        }
-
-        /// <summary>Set the router object identity.</summary>
-        /// <param name="identity">The Glacier2 router's identity.</param>
-        public void SetRouterIdentity(Identity identity)
+        /// <summary>Gets or sets the request context to use while establishing a connection to the Glacier2 router.
```
Comment (externl) [inline csharp/src/Glacier2/SessionFactoryHelper.cs:22]:
```suggestion
        /// <summary>The request context to use while establishing a connection to the Glacier2 router.
```
I'm not sure what our typical pattern is, but I don't see the need to have "Gets or sets" for properties with getters and setters.

File: csharp/src/Glacier2/SessionFactoryHelper.cs [hunk 4]
```diff
@@ -19,180 +19,234 @@ namespace ZeroC.Glacier2
...
+                {
+                    _context = value;
+                }
             }
         }
 
-        /// <summary>Sets the host on which the Glacier2 router runs.</summary>
-        /// <param name="hostname">The host name (or IP address) of the router host.</param>
-        public void SetRouterHost(string hostname)
+        /// <summary>Gets or sets the Glacier2 router port to connect to. If 0, then the default port (4063 for TCP or
```
Comment (externl) [inline csharp/src/Glacier2/SessionFactoryHelper.cs:42]: Ditto
```suggestion
        /// <summary>The Glacier2 router port to connect to. If 0, then the default port (4063 for TCP or
```

File: csharp/src/Glacier2/SessionHelper.cs [hunk 5]
```diff
@@ -4,139 +4,78 @@
...
-
-        /// <summary>
-        /// Destroys the Glacier2 session.
-        ///
-        /// Once the session has been destroyed, SessionCallback.disconnected is
-        /// called on the associated callback object.
-        /// </summary>
-        public void
-        Destroy()
+        /// <summary>Gets the session's communicator object.</summary>
```
Comment (externl) [inline csharp/src/Glacier2/SessionHelper.cs:19]:
```suggestion
        /// <summary>The session's communicator object.</summary>
```

File: csharp/src/Glacier2/SessionHelper.cs [hunk 6]
```diff
@@ -204,47 +153,58 @@ private ObjectAdapter
...
             }
         }
 
-        /// <summary>
-        /// Connects a Glacier2 session using user name and password credentials.
-        ///
-        /// Once the connection is established, SessionCallback.connected is called on the callback object;
-        /// upon failure SessionCallback.exception is called with the exception.
-        /// </summary>
+        /// <summary>Connects a Glacier2 session using user name and password credentials. Once the connection is
```
Comment (externl) [inline csharp/src/Glacier2/SessionHelper.cs:191]:
```suggestion
        /// <summary>Connects a Glacier2 session using username and password credentials. Once the connection is
```

File: csharp/src/Glacier2/SessionFactoryHelper.cs (hunk 3 above)
Comment (pepone) [reply csharp/src/Glacier2/SessionFactoryHelper.cs:22]: Regarding "Gets or sets" I think this is the preferred way for C# properties documentation

see https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1623.md

**Pull request merged**